### PR TITLE
Expanding zero-order water source database

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This PR expands the zero-order database for water sources, and adds some basic tests for the data file.

## Changes proposed in this PR:
- Adds source data for different seawaters, agricultural, municipal, brackish and produced waters.
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/data/techno_economic/water_sources.yml [hunk 1]
```diff
@@ -1,4 +1,312 @@
...
+      constituent_longform: Total Dissolved Solids (TDS)
+      value: 2.5
+      units: kg/m**3
+  default_flow:
+    value: 1.336278
+    units: m**3/s
+  pH:
+    max: 7.6
+    min: 7.6
+brackish_high:
```
Comment (adam-a-a) [inline watertap/data/techno_economic/water_sources.yml:117]: Between `brackish_ave` and `brackish_high`, there isn't a significant difference in TDS. If we want a composition representative of highly brackish water, we should have a set of solutes that corresponds to TDS around 10,000 ppm. Even though the naming of these two probably just follow some specific case provided in the study mentioned in metadata, it isn't representative of highly brackish water. Just something to have in mind for users who may look through these water sources to choose a set representative of highly brackish groundwater.

Comment (adam-a-a) [reply watertap/data/techno_economic/water_sources.yml:117]: Note, the definition of high brackish vs. low/medium can vary by literature source, but I still don't think we'd regard this a high brackish.

**Review by adam-a-a — APPROVED** [review]
LGTM - just a minor comment on the "brackish_high" water source. We can add data for a highly brackish water source later on. I'd probably rename "brackish_high" since we'd want to reserve that name for a highly brackish source.

File: watertap/data/techno_economic/water_sources.yml (hunk 1 above)
Comment (andrewlee94) [reply watertap/data/techno_economic/water_sources.yml:117]: Again - I only had the data from WT3 available and I don't know enough to change it (or judge it). I will also note they had a brackish_low dataset, but that was identical to the brackish_ave set.

Comment (kurbansitterley) [reply watertap/data/techno_economic/water_sources.yml:117]: A better brackish_high could be the san_luis data at ~12 g/L TDS, but we considered that an agriculture source.

**Review by kurbansitterley — COMMENTED** [review]
LGTM, and I echo @adam-a-a comment on brackish_high.

**Review by kurbansitterley — APPROVED** [review]
LGTM, and I echo @adam-a-a comment on brackish_high.

**Pull request merged**